### PR TITLE
Switch modal to composition

### PR DESCRIPTION
**Pull request by ValJed**
## Summary

Since `AposFocus` and `AposFocusMixin` were duplicated, I switched `AposModal` (the only other component using `AposFocusMixin`).

Use reactive state easier to store in a normal window array (a pinia state would be more adapted for this kind of things, like page context).

## What are the specific steps to test this change?

Modals are working without errors,
the last focus element of a modal is re focused when going back to this modal.

## What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] New feature
- [X] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/modal/ui/apos/components/AposModal.vue [hunk 1]
```diff
@@ -94,186 +94,196 @@
...
+    return `${element}:not([tabindex="-1"]):not([disabled]):not([type="hidden"]):not([aria-hidden])`;
+  }
+}
+
+function close() {
+  if (apos.modal.stack.at(-1)?.id !== state.id) {
+    return;
   }
-};
+  emit('esc');
```
Comment (haroun) [inline modules/@apostrophecms/modal/ui/apos/components/AposModal.vue:285]: Are the event listeners still active when I call close directly?

Comment (ValJed) [reply modules/@apostrophecms/modal/ui/apos/components/AposModal.vue:276]: Yes!

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue [hunk 2]
```diff
@@ -405,11 +405,11 @@ export default {
     },
     shortcutNew(event) {
       const interesting = event.keyCode === 78; // N(ew)
-      const topModal = apos.modal.stack[apos.modal.stack.length - 1] ? apos.modal.stack[apos.modal.stack.length - 1].id : null;
+      const topModalId = apos.modal.stack.at(-1)?.id;
```
Comment (boutell) [inline modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:408]: Must start using this I see it's been in node and supported browsers for a while

**Review by boutell — APPROVED** [review]
LGTM

**Review by haroun — APPROVED** [review]

**Pull request merged**